niceman/distributions/conda.py

```diff
@@ -92,13 +92,43 @@ def install_packages(self, session=None):
             Environment sub-class instance.
         """
 
-        # TODO: Need to figure out a graceful way to install conda before we can install packages here.
-        # for package in self.provenance['packages']:
-        #     session.add_command(['conda',
-        #                            'install',
-        #                            package['name']])
+        # TODO: Install Conda
+#        print("We got" + str(environment))
+        # Loop through non-root packages, creating the conda-env export
+        # file to import
+        for env in self.environments:
+            if env.path == self.path:
+                continue
+            export_contents = self.create_conda_export(env)
+            print ("EXPORT\n" + export_contents)
+
         return
 
+    @staticmethod
+    def create_conda_export(env):
+        # TODO: The environment name should be discovered on retrace
+        name = os.path.basename(os.path.normpath(env.path))
+        export = "name: %s\n" % name
+        # Collect channels
+        export += "channels:\n"
+        for c in env.channels:
+            export += "- %s\n" % c["name"]
+        export += "dependencies:\n"
+        # Collect packages (separating pip installed packages)
+        pip_export = ""
+        for p in env.packages:
+            if p.get("installer") is None:
+                export += "- %s=%s=%s\n" % (p["name"], p["version"],
+                                            p["build"])
```

Comment:
Purely a personal preference, but with `.format()`, this could be

```python
export += "- {name}={version}={build}".format(**p)
```

Reply:
IMHO it should check `version` and `build` on being defined.   would it tollerate `None` or should they be provided an alternative -- an empty string (i.e. `or ""`) so if not defined -- it might then look like `- pkg==` (is that legit?)

Reply:
OK, this has led me to delve into the conda-env code: https://github.com/conda/conda-env/blob/develop/conda_env/env.py .  Clearly I can learn some lessons from them (making the file exactly as they do).  It does look however that pkg== is legit from their construction, but I'll test it out :)

Reply:
OK, pkg== is an invalid package specification. So I need to be more nuanced in building the package name.  (Time for a helper function.)